### PR TITLE
Shorten inline helptext for ecommerce checkbox, #PG-3647

### PR DESCRIPTION
**Pull request by adamreng**
### Description:

This PR addresses the overly long description for the "Is Ecommerce View" checkbox when creating or editing a Matomo Analytics tag with Pageview tracking type. The changes include:

1. Shortening the existing description text to improve readability.
2. Implementing conditional text based on whether the setup is cloud-based or on-premises.

Addresses PG-3647

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: Template/Tag/MatomoTag.php [hunk 1]
```diff
@@ -51,7 +52,11 @@ public function getParameters()
...
         $isEcommerceView = $this->makeSetting('isEcommerceView', false, FieldConfig::TYPE_BOOL, function (FieldConfig $field) {
             $field->title = Piwik::translate('TagManager_MatomoTagEcommerceViewIsEcommerceView');
-            $field->description = Piwik::translate('TagManager_MatomoTagEcommerceViewIsEcommerceViewHelp');
+            $field->inlineHelp = '';
+            if (!Manager::getInstance()->isPluginActivated('Cloud')) {
+                $field->inlineHelp .= Piwik::translate('TagManager_MatomoTagEcommerceViewIsEcommerceViewHelpOnPrem') . '<br /><br />';
+            }
+            $field->inlineHelp .= Piwik::translate('TagManager_MatomoTagEcommerceViewIsEcommerceViewHelpV2');
             $field->condition = 'trackingType == "pageview"';
         });
```
Comment (araichyk) [inline Template/Tag/MatomoTag.php:61]: Let me know if there's a better way to determine if the environment is cloud or on-prem. But I tested locally with activated/deactivated Cloud plugin and it seems to work.

File: Template/Tag/MatomoTag.php [hunk 2]
```diff
@@ -51,7 +52,11 @@ public function getParameters()
...
         $isEcommerceView = $this->makeSetting('isEcommerceView', false, FieldConfig::TYPE_BOOL, function (FieldConfig $field) {
             $field->title = Piwik::translate('TagManager_MatomoTagEcommerceViewIsEcommerceView');
-            $field->description = Piwik::translate('TagManager_MatomoTagEcommerceViewIsEcommerceViewHelp');
+            $field->inlineHelp = '';
+            if (!Manager::getInstance()->isPluginActivated('Cloud')) {
```
Comment (araichyk) [inline Template/Tag/MatomoTag.php:56]: Let me know if there's a better way to determine if the environment is cloud or on-prem. But I tested locally with activated/deactivated Cloud plugin and it seems to work.

File: Template/Tag/MatomoTag.php (hunk 1 above)
Comment (adamreng) [inline Template/Tag/MatomoTag.php]: 1. Let me know if there's a better way to determine if the environment is cloud or on-prem than this `isPluginActivated('Cloud')` check. I tested locally with the Cloud plugin both activated and deactivated and it seems to work.
2. Changing from `$this->description` to `$this->inlineHelp` is to allow the `<br />` to be handled as HTML. I don't think there are any negative effects of that change, but I'm not 100% sure. If there are, let me know.

**Review by snake14 — APPROVED** [review]
Looks good to me. I confirmed that the text changes when the Cloud plugin is activated.

When changing behaviour for Cloud, we often use dependency injection (StaticContainer), but since you are only changing part of the text for Cloud, I think this approach works fine.

**Review by AltamashShaikh — APPROVED** [review]
@araichyk A more cleaner way to do this is via plugins `config.php`, the current implementation also works for me :+1: 

In `TagManager/config/config.php` add this line
```
'TagManager.MatomoTagEcommerceViewIsEcommerceViewHelpOnPremise' => Piwik\DI::string('TagManager_MatomoTagEcommerceViewIsEcommerceViewHelpOnPrem'),
```

In `Cloud/config/config.php` add this line to make the first line as empty
```
'TagManager.MatomoTagEcommerceViewIsEcommerceViewHelpOnPremise' => '',
```

Now you can use it in following way in `TagManager/Template/Tag/MatomoTag.php`
```
$inLinHelpLine1 = StaticContainer::get('TagManager.MatomoTagEcommerceViewIsEcommerceViewHelpOnPremise');
$field->inlineHelp = ($inLinHelpLine1 ? Piwik::translate($inLinHelpLine1) . '<br><br>' : '') . Piwik::translate('TagManager_MatomoTagEcommerceViewIsEcommerceViewHelpV2');
```

File: Template/Tag/MatomoTag.php (hunk 1 above)
Comment (AltamashShaikh) [reply Template/Tag/MatomoTag.php]: @araichyk inlineHelp is the way to go if description has HTML tags :+1:

**Comment by adamreng** [comment]
@AltamashShaikh Thank you for explaining that in detail! I've implemented the changes as suggested and pushed them up. They seem to work correctly in my local environment.

I've also created a corresponding PR for the Cloud plugin: https://github.com/innocraft/plugin-Cloud/pull/905.

**Review by AltamashShaikh — APPROVED** [review]
@araichyk Thanks, looks good :+1:

**Pull request merged**